t/unit/transport/test_redis.py

```diff
@@ -1,22 +1,35 @@
 from __future__ import absolute_import, unicode_literals
 
+import fakeredis
 import pytest
+import redis
+import unittest
 import socket
-import types
 
+from array import array
+from case import ANY, ContextMock, Mock, call, mock, skip, patch
 from collections import defaultdict
+from contextlib import contextmanager
 from itertools import count
 
-from case import ANY, ContextMock, Mock, call, mock, skip, patch
-
 from kombu import Connection, Exchange, Queue, Consumer, Producer
 from kombu.exceptions import InconsistencyError, VersionMismatch
-from kombu.five import Empty, Queue as _Queue, bytes_if_py2
+from kombu.five import Empty, Queue as _Queue
 from kombu.transport import virtual
 from kombu.utils import eventio  # patch poll
 from kombu.utils.json import dumps
 
 
+_fake_redis_client = None
+
+
+def _get_fake_redis_client():
+    global _fake_redis_client
+    if _fake_redis_client is None:
+        _fake_redis_client = FakeRedisClient()
+    return _fake_redis_client
```

Comment:
While this implies you'd want to use the fake redis client as a singleton -- set up once and then used forever -- in actual use in the test suite, you take care to destroy the global in `tearDown()` functions, to force it to be set up anew later. This seems wrong -- it looks like you should be using normal objects, perhaps setting the client as an attribute of the TestCase instance, rather than try to make it a singleton fetched through a function and then "reach behind the function's back" to manipulate the global.